src/features/device-heartbeat/index.ts

```diff
@@ -324,29 +330,36 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 		return null;
 	}
 
+	private async getDeviceOnlineState(deviceId: number) {
+		try {
+			const value = await redisRO.get(
+				`${DeviceOnlineStateManager.REDIS_NAMESPACE}:${deviceId}`,
+			);
+			if (value == null) {
+				return;
+			}
+			return JSON.parse(value) as DeviceOnlineStateManagerMessage;
+		} catch {
+			// Ignore
+		}
+	}
+
 	private async scheduleChangeOfStateForDevice(
 		deviceId: number,
+		currentManagerState: DeviceOnlineStateManagerMessage | undefined,
```

Comment:
I think it'd be a bit clearer as:
```suggestion
		previousManagerState: DeviceOnlineStateManagerMessage | undefined,
```